src/device.rs

```diff
@@ -155,6 +159,46 @@ impl From<wgt::Limits> for CLimits {
     }
 }
 
+#[repr(u8)]
+pub enum CDeviceType {
+    /// Other.
+    Other = 0,
+    /// Integrated GPU with shared CPU/GPU memory.
+    IntegratedGpu,
+    /// Discrete GPU with separate CPU/GPU memory.
+    DiscreteGpu,
+    /// Virtual / Hosted.
+    VirtualGpu,
+    /// Cpu / Software Rendering.
+    Cpu,
+}
+
+impl From<DeviceType> for CDeviceType {
+    fn from(other: DeviceType) -> Self {
+        match other {
+            DeviceType::Other => CDeviceType::Other,
+            DeviceType::IntegratedGpu => CDeviceType::IntegratedGpu,
+            DeviceType::DiscreteGpu => CDeviceType::DiscreteGpu,
+            DeviceType::VirtualGpu => CDeviceType::VirtualGpu,
+            DeviceType::Cpu => CDeviceType::Cpu,
+        }
+    }
+}
+
+#[repr(C)]
+pub struct CAdapterInfo {
+    /// Adapter name
+    pub name: *mut c_char,
```

Comment:
we need the user providing `name_length` variable, and then us using that much data and changing it